### PR TITLE
Fix false positive when the node is the only one inside of if/else branch

### LINKED ISSUE #648: RSpec/EmptyLineAfterExampleGroup invalid behavior
When I have a conditional spec based on some external arguments EmptyLine linter requires a new line after the spec but this does not seem right. I use flags like this sometimes for extensive integration testing of some internal components that I want to run only on special occasions. For such cases, this linter shouldn't require a new line as it doesn't make sense.

```ruby
describe Example do
  if ENV['FLAG']
    describe 'something funny' do
      let (:name) { 45 }

      it 'whatever' do
      end
    end
  end
end
```

Please use this code for reproduction:

```
git clone https://github.com/mensfeld/broken-rubocop-rspec4.git
cd broke-rubocop-rspec4
./run.sh
```

Result:

```
Inspecting 2 files
.C

Offenses:

spec/example_spec.rb:8:5: C: RSpec/EmptyLineAfterExampleGroup: Add an empty line after describe.
    end
    ^^^
```

### PR DESCRIPTION
**Pull request by Darhazer**
Fixes #648

---

Before submitting the PR make sure the following are checked:

* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [changelog](https://github.com/rubocop-hq/rubocop-rspec/blob/master/CHANGELOG.md) if the new code introduces user-observable changes.
* [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

If you have created a new cop:

* [ ] Added the new cop to `config/default.yml`.
* [ ] The cop documents examples of good and bad code.
* [ ] The tests assert both that bad code is reported and that good code is not reported.

### CODE REVIEW

File: lib/rubocop/rspec/blank_line_separation.rb [hunk 1]
```diff
@@ -25,6 +25,16 @@ def offending_loc(last_line)
         source_range(processed_source.buffer, last_line, start, content_length)
       end
 
+      def single_node?(node)
+        return if node.parent.nil?
+
+        if node.parent.if_type?
```
Comment (dgollahon) [inline lib/rubocop/rspec/blank_line_separation.rb]: Is there a reason you've chosen to solve for only the if type explicitly? I imagine there are other constructs that would have this same problem (my first thought is `case`, but i possibly also `while` or `for` loops or something i'm not thinking of)

Would @bquorning's thought about checking if it's in a block/begin node not work https://github.com/rubocop-hq/rubocop-rspec/issues/648#issuecomment-397804501 ?

Comment (Darhazer) [reply lib/rubocop/rspec/blank_line_separation.rb]: You are right, we have to check `begin` nodes. I even don't see why a block node have to be checked - if there is more then one child of a block, it have to be wrapped in a begin.

Also, it's not single_node, but last_child

PR updated

**Comment by Darhazer** [comment]
@bquorning it seems that you've reviewed the old version of the diff. It's already called `last_child?` and doesn't return any `nil` :)

File: lib/rubocop/rspec/blank_line_separation.rb [hunk 2]
```diff
@@ -25,6 +25,16 @@ def offending_loc(last_line)
         source_range(processed_source.buffer, last_line, start, content_length)
       end
 
+      def single_node?(node)
+        return if node.parent.nil?
```
[the remark this answers is not in the document]
Comment (Darhazer) [reply lib/rubocop/rspec/blank_line_separation.rb]: It would be a good cop

**Pull request merged**